### PR TITLE
add names of the provider nodes to the logs and output

### PR DESCRIPTION
**Pull request by shadeofblue**

### CODE REVIEW

**Review by lucekdudek — APPROVED** [review]

File: ray_on_golem/server/services/golem/golem.py [hunk 1]
```diff
@@ -176,17 +176,24 @@ def _score_with_provider_data(
...
         self, context: WorkContext, ip: str, *, add_state_log: Callable[[str], Awaitable[None]]
     ):
-        logger.info(f"Deploying image on `{context._activity}`")
+        activity = context.activity
+        provider_desc = await self._get_provider_desc(context)
+
+        logger.info(f"Deploying image on {provider_desc}, {ip=}, {activity=}")
 
-        await add_state_log("[4/9] Deploying image on activity...")
+        await add_state_log(f"[4/9] Deploying image...")
```
Comment (approxit) [inline ray_on_golem/server/services/golem/golem.py:191]: `f` is not needed.

File: ray_on_golem/server/services/golem/golem.py [hunk 2]
```diff
@@ -209,15 +217,18 @@ async def _upload_node_configuration(
...
 
     async def _start_ssh_server(
-        self, context: WorkContext, *, add_state_log: Callable[[str], Awaitable[None]]
+        self, context: WorkContext, ip: str, *, add_state_log: Callable[[str], Awaitable[None]]
     ):
-        logger.info(f"Starting ssh service on `{context._activity}`")
-        await add_state_log("[7/9] Starting ssh service on activity...")
+        logger.info(
+            "Starting ssh service on "
+            f"{await self._get_provider_desc(context)}, {ip=}, {context.activity=}"
```
Comment (approxit) [inline ray_on_golem/server/services/golem/golem.py:224]: Using `await` expressions inside of f-string is too much for my taste. It's better to have separate variable for readability.

**Review by approxit — CHANGES_REQUESTED** [review]

File: ray_on_golem/server/services/golem/golem.py (hunk 2 above)
Comment (shadeofblue) [reply ray_on_golem/server/services/golem/golem.py:224]: is this really about _tastes_ now? :(

Comment (shadeofblue) [reply ray_on_golem/server/services/golem/golem.py]: but so be it ¯\_(ツ)_/¯

File: ray_on_golem/server/services/golem/golem.py (hunk 1 above)
Comment (shadeofblue) [reply ray_on_golem/server/services/golem/golem.py]: updated

**Review by approxit — APPROVED** [review]

**Pull request merged**